tekton/ci-workspace/jobs/tekton-catalog-diff-task.yaml

```diff
@@ -0,0 +1,110 @@
+apiVersion: tekton.dev/v1beta1
+kind: Pipeline
+metadata:
+  name: diff-task-job
+spec:
+  workspaces:
+    - name: source
+      description: Workspace where the git repo is prepared for linting.
+  params:
+    - name: pullRequestUrl
+      description: The HTML URL for the pull request
+    - name: gitRepository
+      description: The git repository that hosts context and Dockerfile
+    - name: pullRequestBaseRef
+      description: The pull request base branch
+    - name: pullRequestNumber
+      description: The pullRequestNumber
+  tasks:
+    - name: clone-repo
+      taskRef:
+        name: git-clone
+        bundle: gcr.io/tekton-releases/catalog/upstream/git-clone:0.4
+      workspaces:
+        - name: output
+          workspace: source
+      params:
+        - name: url
+          value: $(params.gitRepository)
+        - name: revision
+          value: $(params.pullRequestBaseRef)
+        - name: subdirectory
+          value: ""
+        - name: deleteExisting
+          value: "true"
+    - name: get-task-diff
+      runAfter:
+        - clone-repo
+      taskRef:
+        name: git-cli
+        bundle: gcr.io/tekton-releases/catalog/upstream/git-cli:0.3
+      workspaces:
+        - name: source
+          workspace: source
+      params:
+        - name: GIT_SCRIPT
+          value: |
+            # This needs to be set or the pipeline will fail on the next task.
+            echo -n "" > $(results.commit.path)
+
+            temp_branch="temp-branch-pr-$(params.pullRequestNumber)"
+            git fetch origin pull/$(params.pullRequestNumber)/head:${temp_branch}
+            git checkout ${temp_branch}
```

Comment:
i think you could have just git reset --hard FETCH_HEAD without going by temp_branch but that work both way!